harvdev_utils/psycopg_functions/fb_feature_classes.py

```diff
@@ -498,10 +498,138 @@ class Insertion(Feature):
+        self.chr_id = None         # Will be feature_id for Dmel chr/scaffold (golden_path(_region)).
+        self.fmin = None           # Will be featureloc.fmin (interbase).
+        self.fmax = None           # Will be featureloc.fmax (interbase).
+        self.strand = None         # Will be featureloc.strand.
+        self.agr_start = None      # Will be onbase "start" for export (adjusted as needed).
+        self.agr_stop = None       # Will be onbase "stop" for export (adjusted as needed).
+        # Attributes below require reference assembly info for determination.
+        self.chr_name = None       # Will be "uniquename" of the chromosome/scaffold.
+        self.chr_acc = None        # Will be the REFSEQ accession.version for the chromosome/scaffold.
+        self.ref_seq = None        # Will be the starting genomic sequence, usually "N/A" (string).
+        self.alt_seq = None        # Will be the variant genomic sequence, if known (string).
+        self.alt_acc = None        # Will be accession for inserted sequence, db:acc (string).
+        self.ins_len = None        # Will be the size of the insertion (integer).
+        self.padded_base = None    # Will be a string: A, C, G or T.
+        # Attributes below require synthesis of various Insertion data.
+        self.fails = []            # Will list various reasons object couldn't be exported.
+        self.agr_type = None       # Will be a Sequence Ontology term ID: e.g., usually "SO:0000667".
+        self.for_agr_export = None # Will be boolean.
 
     # feature.uniquename must be FBti-type.
     uniquename_regex = r'^FBti[0-9]{7}$'
 
+    def pick_floc(self):
+        """Determine featureloc specifics only if one featureloc entry found."""
+        if type(self.floc_list) is None:
+            log.warning('The "floc_list" is None - expected a list (even if empty).')
+        elif type(self.floc_list) != list:
+            log.warning('The "floc_list" is not a list - expected a list (even if empty).')
+        elif len(self.floc_list) == 0:
+            log.debug('Can\'t determine feature location for {}: no featureloc info in chado.'.format(self.uniquename))
+        elif len(self.floc_list) > 1:
+            log.debug('Can\'t determine feature location for {}: many featureloc values in chado.'.format(self.uniquename))
+        else:
+            self.chr_id = self.floc_list[0][0]
+            self.fmin = self.floc_list[0][1]
+            self.fmax = self.floc_list[0][2]
+            self.strand = self.floc_list[0][3]
+            if type(self.fbrf_list) == list and re.match(r'^FBrf[0-9]{7}$', str(self.floc_list[0][4])):
+                self.fbrf_list.append(self.floc_list[0][4])
+
+        return
+
+    def get_agr_floc(self):
+        """Determine onbase start/stop coordinates and from that, distinguish insertion from delin."""
+        # First make sure necessary floc info is present.
+        if self.fmin is None:
```

Comment:
Thats a lot of if else, would be nice to have as function calls i.e.

def teis():
   do something

def is():
   do something else

process_dict = {
   'transposable_element_insertion_site': teis,
   'insertion_site': is
}

if bob in process_dict:
    process_dict[bob]()

Does that make sense ?

Reply:
Sorry having problems with the formatting..... grrr.

Reply:
Great suggestion. Thanks!